src/cmd-test-coreos-installer

```diff
@@ -0,0 +1,120 @@
+        if test -s ${tmpf_snap}; then       # py3's http.server prints the http:// address also
+          sed -e 's,Serving HTTP on 0.0.0.0 port \([0-9]*\)\( (http://[^)]*)\)\? \.\.\.,\1,' < ${tmpf_snap} > ${statedir}/httpd-port
+            if ! cmp ${tmpf_snap} ${statedir}/httpd-port 1>/dev/null; then
+                # If so, we've successfully extracted the port
+                break
+            fi
+        fi
+        sleep 1
+    done
+
+    if [ ! -f ${statedir}/httpd-port ]; then
+      cat ${statedir}/httpd-output
+      echo "can't start up httpd"; exit 1
+    fi
+
+    port=$(cat ${statedir}/httpd-port)
+    echo "http://127.0.0.1:${port}" > ${statedir}/httpd-address
+    echo "$child_pid" > ${statedir}/httpd-pid
+}
+
+set -x
+
+build=${1:-latest}
+
+builddir=builds/${build}/x86_64
+buildmeta=${builddir}/meta.json
+
+buildid=$(jq -er .buildid < ${buildmeta})
+metalimg=$(jq -er .images.metal.path < ${buildmeta})
+instkernel=$(jq -er .images.kernel.path < ${buildmeta})
+instinitramfs=$(jq -er .images.initramfs.path < ${buildmeta})
+
+# OK, we seem to have the images.  Launch our temporary webserver.
+statedir=tmp/coreos-installer-test
+rm -rf "${statedir}" && mkdir -p "${statedir}"
+tftpdir=${statedir}/tftp
+pxeconfigdir=${tftpdir}/pxelinux.cfg
+mkdir -p "${pxeconfigdir}"
+
+run_tmp_webserver "${statedir}" "${tftpdir}"
+port=$(cat ${statedir}/httpd-port)
+
+case "${metalimg}" in
+  *.raw) echo "NOTICE: Metal image is not compressed, doing so now temporarily"
+         gzip -1 < "${builddir}/${metalimg}" > "${tftpdir}/${metalimg}.gz"
+         metalimg=${metalimg}.gz ;;
+  *) ln "${builddir}/${metalimg}" "${tftpdir}" ;;
+esac
+
+for x in ${instkernel} ${instinitramfs}; do
+    ln "${builddir}/${x}" "${tftpdir}"
+done
+cat > ${pxeconfigdir}/default << EOF
+DEFAULT pxeboot
+TIMEOUT 20
+PROMPT 0
+LABEL pxeboot
+    KERNEL ${instkernel}
+    APPEND ip=dhcp rd.neednet=1 initrd=${instinitramfs} console=tty0 console=ttyS0 coreos.inst=yes coreos.inst.install_dev=vda coreos.inst.image_url=http://192.168.76.2:${port}/${metalimg} coreos.inst.ignition_url=http://192.168.76.2:${port}/config.ign
+IPAPPEND 2
+EOF
+
+cp --reflink=auto /usr/share/syslinux/{pxelinux.0,ldlinux.c32} ${tftpdir}
+
+disk=${statedir}/disk.qcow2
+qemu-img create -f qcow2 ${disk} 12G
+
+cat > ${tftpdir}/config.ign << EOF
+{
+  "ignition": { "version": "3.0.0" },
+  "systemd": {
+    "units": [{
+      "name": "coreos-test-installer.service",
+      "enabled": true,
+      "contents": "[Unit]\nRequires=dev-virtio\\\x2dports-completion.device\nOnFailure=emergency.target\nOnFailureJobMode=isolate\n[Service]\nType=oneshot\nExecStart=/bin/sh -c '/usr/bin/echo coreos-installer-test-OK >/dev/virtio-ports/completion && systemctl poweroff'\n\n[Install]\nRequiredBy=multi-user.target"
+    }]
+  }
+}
+EOF
+jq . < ${tftpdir}/config.ign >/dev/null
+
+completionf=${statedir}/completion.txt
+touch "${completionf}"
+qemu-system-x86_64 -accel kvm -m 8192 -nographic \
```

Comment:
Greedy :) 

This might be a problem if we want to plumb it into the pipelines.